### PR TITLE
New module: Fusioninspector

### PR DESCRIPTION
**Pull request by delfiterradas**
## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [ ] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

File: modules/nf-core/fusioninspector/main.nf [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+    FusionInspector \\
+        --fusions $fusion_list \\
+        --genome_lib ${reference} \\
+        $fasta \\
+        --CPU ${task.cpus} \\
+        -O . \\
+        --out_prefix $prefix \\
+        --vis $args $args2
+
+    # Touch the output files to make sure they exist
```
Comment (itrujnara) [inline modules/nf-core/fusioninspector/main.nf]: Is there some logic behind which files/directories are created? It would be great to be more principled with this if we can determine which outputs we expect. You could mark more output channels as optional or add some more ternaries

File: modules/nf-core/fusioninspector/main.nf [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://community-cr-prod.seqera.io/docker/registry/v2/blobs/sha256/13/139b94a1f10c6e23a8c27eaed1e5a689db978a513d0ee155e74d35f0970814fe/data' :
+        'community.wave.seqera.io/library/fusion-inspector_igv-reports_perl-json-xs_pysam_pruned:c6147971d107ab11'}"
+
+    input:
+    tuple val(meta), path(reads), path(fusion_list)
+    tuple val(meta2), path(reference)
+
+    output:
+    tuple val(meta), path("*FusionInspector.fusions.tsv")   , emit: tsv
```
Comment (itrujnara) [inline modules/nf-core/fusioninspector/main.nf]: You can move the commas closer, to just align with the longest tuple

File: modules/nf-core/fusioninspector/main.nf [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+    tuple val(meta), path("IGV_inputs")                     , emit: igv_inputs
+    tuple val(meta), path("fi_workdir")                     , emit: fi_workdir
+    tuple val(meta), path("chckpts_dir")                    , emit: chckpts_dir
+    path "versions.yml"                                     , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def prefix = task.ext.prefix ?: "${meta.id}"
```
Comment (itrujnara) [inline modules/nf-core/fusioninspector/main.nf:29]: Harshil alignment

File: modules/nf-core/fusioninspector/main.nf [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        FusionInspector: \$(FusionInspector --version 2>&1 | grep -i 'version' | sed -e 's/FusionInspector version: //' -e 's/[[:space:]]//g')
+    END_VERSIONS
+    """
+
+    stub:
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    """
+    touch FusionInspector.log
```
Comment (itrujnara) [inline modules/nf-core/fusioninspector/main.nf:52]: Is this the basic output of the tool? If so, it's long enough to consider a template file

File: modules/nf-core/fusioninspector/tests/main.nf.test [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+    name "Test Process FUSIONINSPECTOR"
+    script "../main.nf"
+    process "FUSIONINSPECTOR"
+    tag "modules"
+    tag "modules_local"
+    tag "fusioninspector"
+    tag "starfusion/build"
+    tag "modules_nfcore"
+
+    test("FUSIONINSPECTOR - test") {
```
Comment (itrujnara) [inline modules/nf-core/fusioninspector/tests/main.nf.test:12]: I am not familiar with the tool at all, is there any alternative input/mode that would need testing? Fine if there's none

**Review by itrujnara — COMMENTED** [review]
Hi, very nice PR, almost ready to go. I've left a few remarks to consider, plus some formatting comments.
